### PR TITLE
pattern option to provide a sample rate to sample logs based on pseud…

### LINKED ISSUE #4: Sample log patterns
Currently, you can only preserve or discard logs. It would be nice to have a random sample in the event that you want to care about something but not every entry of it.

Something like:
```
- regex: 'Waited for .* due to client-side throttling'
  sample: 0.01 # keep 1% of it (can be random)
```

### PR DESCRIPTION
**Pull request by a7i**
…o-random

closed #4

### CODE REVIEW

**Comment by a7i** [comment]
cc: @grosser

File: config.go [hunk 1]
```diff
@@ -65,6 +68,13 @@ func NewConfig(path string) (*Config, error) {
 		config.Patterns[i].regexParsed =
 			helpfulMustCompile(config.Patterns[i].Regex, "patterns["+strconv.Itoa(i)+"].regex")
 		config.Patterns[i].levelSet = (config.Patterns[i].Level != "")
+
+		if config.Patterns[i].SampleRate != nil {
+			rate := *config.Patterns[i].SampleRate
+			if rate < 0.0 || rate > 1.0 {
+				return nil, fmt.Errorf("sample must be between 0.0 - 1.0 but was %f", rate)
```
Comment (grosser) [inline config.go:75]: ❤️

File: main.go [hunk 2]
```diff
@@ -31,6 +32,8 @@ func main() {
 		defer config.Statsd.Stop()
 	}
 
+	rand.Seed(time.Now().UnixNano())
```
Comment (grosser) [inline main.go:35]: some comment on what this does

File: config_test.go [hunk 3]
```diff
@@ -25,5 +27,17 @@ var _ = Describe("config", func() {
 				Expect(err).ToNot(BeNil())
 			})
 		})
+
+		It("fails on invalid sample rate", func() {
```
Comment (grosser) [inline config_test.go:31]: add a positive test with a valid rate ?

Comment (grosser) [reply config_test.go:31]: nvm .. rate is tested below already

**Review by grosser — APPROVED** [review]

**Pull request merged**